Chart js2.8

mahugnon opened the pull request:
We propose an update of ChartJs to 2.8 version of ChartJS

jecisc committed:
Merge branch 'addInfoOnReleasev1'

jecisc committed:
Merge pull request #11 from DuneSt/correctDemoLink

Correct demo link

jecisc committed:
Merge pull request #12 from DuneSt/development

Release v1.0.1: Use stylesheet from github

jecisc committed:
Merge pull request #13 from DuneSt/development

v1.0.2

mahugnon committed:
Start rebuilding chartJs library

mahugnon committed:
draw chart ChartJs 2.0

mahugnon committed:
remove fil

mahugnon committed:
add option. Fill
DataSets setOption: anOption to: value

mahugnon committed:
add a strategy to set options

mahugnon committed:
Unified Chart1dData and Chart2dData
Set stylesheet default colors
Change forChartJs to convert pharo colors to rgb

mahugnon committed:
Remove ChartOptions
Rename Char2dData to ChartData

mahugnon committed:
Default option by chartType

mahugnon committed:
Merge a349b12a35d0fb8e272303b5248c8472a1d4c944

mahugnon committed:
Rmove Tooltips options
Remove unused methods

jecisc commented:
I opened the demo and saw some problems:
- The 1D demos do not have any color
- The Polar shape does not print anything
- The BoxPlot does not print anything
- The style demo is broken
- The callback feature is broken

This need to be addressed before we review the model.

mahugnon commented:
Yes, you're right Cyril.  It has to be improved.   We just worked on 2D plots and little bit 1D plots

jecisc committed:
Merge branch 'development'

jecisc committed:
Merge branch 'development'

jecisc committed:
Merge branch 'development'

mahugnon committed:
testing and hacking

mahugnon committed:
make ChartJs 1d work

mahugnon committed:
update ChartJS:
-create chartType1d to check if the chart is one dimenssion 
-resetBackgroundwithDataSize transform the background color to an array as 1d charts  have array as background color

mahugnon commented:
ChartJs 2.8 does not support BoxPlot directly

mahugnon committed:
Seem to work .
add accessor using set option
support block as color

mahugnon committed:
Boxplot is working

mahugnon committed:
Added Violin chart with demo

mahugnon committed:
fixing callback

mahugnon committed:
Handle call backs

mahugnon committed:
fix tests

mahugnon committed:
fix baseline conflict

jecisc committed:
Merge branch 'master' into ChartJS2.8

jecisc commented:
I tried the demo and now it does not open anymore.

Here is the error I get:

```
UndefinedObject>>doesNotUnderstand: #new
ChartData>>resetBackgroundwithDataSize
[ :jsonMapWriter | 
jsonMapWriter writeKey: #labels value: (xObjects collect: [ :aXObject | self labelFor: aXObject ]).
jsonMapWriter writeKey: #datasets value: self resetBackgroundwithDataSize ] in ChartData>>neoJsonOn: in Block: [ :jsonMapWriter | ...
[ block value: mapWriter ] in NeoJSONWriter>>writeStreamingDo: in Block: [ block value: mapWriter ]
NeoJSONWriter>>indentedDo:
NeoJSONWriter>>writeStreamingDo:
NeoJSONWriter>>writeMapStreamingDo:
ChartData>>neoJsonOn:
NeoJSONWriter>>nextPut:
ChartJsComponent>>printDiagramCreationOn:inDiv:
[ :s | 
self printDiagramContextCreationOn: s inDiv: divId.
s cr.
self printDiagramCreationOn: s inDiv: divId ] in [ self data hasData
	ifTrue: [ html div
			class: #chartJsGraph;
			with: [ canvas := html canvas id: #graph , divId.
				self manageCallBackIn: html canvas: canvas inDiv: divId ].
		self createIfNecessaryLegend: html inDiv: divId.
		html
			script:
				(String
					streamContents: [ :s | 
						self printDiagramContextCreationOn: s inDiv: divId.
						s cr.
						self printDiagramCreationOn: s inDiv: divId ]) ]
	ifFalse: [ html div
			class: 'mdl-alert mdl-alert__danger';
			style: 'color: red; text-align: center; margin: auto;';
			with: [ html text: 'No data to display' ] ].
self data hasTitle ifTrue: [ self renderTitleId: divId on: html ] ] in ChartJsComponent>>renderContentOn: in Block: [ :s | ...
String class(SequenceableCollection class)>>new:streamContents:
String class(SequenceableCollection class)>>streamContents:
[ self data hasData
	ifTrue: [ html div
			class: #chartJsGraph;
			with: [ canvas := html canvas id: #graph , divId.
				self manageCallBackIn: html canvas: canvas inDiv: divId ].
		self createIfNecessaryLegend: html inDiv: divId.
		html
			script:
				(String
					streamContents: [ :s | 
						self printDiagramContextCreationOn: s inDiv: divId.
						s cr.
						self printDiagramCreationOn: s inDiv: divId ]) ]
	ifFalse: [ html div
			class: 'mdl-alert mdl-alert__danger';
			style: 'color: red; text-align: center; margin: auto;';
			with: [ html text: 'No data to display' ] ].
self data hasTitle ifTrue: [ self renderTitleId: divId on: html ] ] in ChartJsComponent>>renderContentOn: in Block: [ self data hasData...
BlockClosure>>renderOn:

```

mahugnon commented:
Bizare. I will look at that

Le jeu. 2 janv. 2020 à 11:44, CyrilFerlicot <notifications@github.com> a
écrit :

> I tried the demo and now it does not open anymore.
>
> Here is the error I get:
>
> UndefinedObject>>doesNotUnderstand: #new
> ChartData>>resetBackgroundwithDataSize
> [ :jsonMapWriter |
> jsonMapWriter writeKey: #labels value: (xObjects collect: [ :aXObject | self labelFor: aXObject ]).
> jsonMapWriter writeKey: #datasets value: self resetBackgroundwithDataSize ] in ChartData>>neoJsonOn: in Block: [ :jsonMapWriter | ...
> [ block value: mapWriter ] in NeoJSONWriter>>writeStreamingDo: in Block: [ block value: mapWriter ]
> NeoJSONWriter>>indentedDo:
> NeoJSONWriter>>writeStreamingDo:
> NeoJSONWriter>>writeMapStreamingDo:
> ChartData>>neoJsonOn:
> NeoJSONWriter>>nextPut:
> ChartJsComponent>>printDiagramCreationOn:inDiv:
> [ :s |
> self printDiagramContextCreationOn: s inDiv: divId.
> s cr.
> self printDiagramCreationOn: s inDiv: divId ] in [ self data hasData
> 	ifTrue: [ html div
> 			class: #chartJsGraph;
> 			with: [ canvas := html canvas id: #graph , divId.
> 				self manageCallBackIn: html canvas: canvas inDiv: divId ].
> 		self createIfNecessaryLegend: html inDiv: divId.
> 		html
> 			script:
> 				(String
> 					streamContents: [ :s |
> 						self printDiagramContextCreationOn: s inDiv: divId.
> 						s cr.
> 						self printDiagramCreationOn: s inDiv: divId ]) ]
> 	ifFalse: [ html div
> 			class: 'mdl-alert mdl-alert__danger';
> 			style: 'color: red; text-align: center; margin: auto;';
> 			with: [ html text: 'No data to display' ] ].
> self data hasTitle ifTrue: [ self renderTitleId: divId on: html ] ] in ChartJsComponent>>renderContentOn: in Block: [ :s | ...
> String class(SequenceableCollection class)>>new:streamContents:
> String class(SequenceableCollection class)>>streamContents:
> [ self data hasData
> 	ifTrue: [ html div
> 			class: #chartJsGraph;
> 			with: [ canvas := html canvas id: #graph , divId.
> 				self manageCallBackIn: html canvas: canvas inDiv: divId ].
> 		self createIfNecessaryLegend: html inDiv: divId.
> 		html
> 			script:
> 				(String
> 					streamContents: [ :s |
> 						self printDiagramContextCreationOn: s inDiv: divId.
> 						s cr.
> 						self printDiagramCreationOn: s inDiv: divId ]) ]
> 	ifFalse: [ html div
> 			class: 'mdl-alert mdl-alert__danger';
> 			style: 'color: red; text-align: center; margin: auto;';
> 			with: [ html text: 'No data to display' ] ].
> self data hasTitle ifTrue: [ self renderTitleId: divId on: html ] ] in ChartJsComponent>>renderContentOn: in Block: [ self data hasData...
> BlockClosure>>renderOn:
>
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/DuneSt/ChartJs/pull/15?email_source=notifications&email_token=AF7H2SZHJAHMEKFLMNUQT7DQ3XASBA5CNFSM4JDB4JFKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEH6CJ2I#issuecomment-570172649>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AF7H2SZSUNQLIQMRNFTSDRLQ3XASBANCNFSM4JDB4JFA>
> .
>

mahugnon committed:
Change RTCPQualitative to color wheel

mahugnon committed:
Merge edc723de1dd6415134cc940bff84eeb1e070aaf6
